### PR TITLE
Make tservice functions explicit

### PR DESCRIPTION
**Pull request by nick-mobilecoin**
Also adds READMEs for the tservice crates

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #47** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/47" target="_blank">Graphite</a> 
    * **PR #50** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/50" target="_blank">Graphite</a> 
    * **PR #51** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/51" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/51?utm_source=stack-comment).

File: tservice/sys/types/build.rs [hunk 1]
```diff
@@ -2,51 +2,39 @@
...
+    "_sgx_dh_session_enclave_identity_t",
+    "_sgx_dh_session_role_t",
+    "_sgx_dh_session_t",
+    "_sgx_report2_mac_struct_t",
+    "_tee_cpu_svn_t",
+    "_tee_measurement_t",
+    "_tee_report_data_t",
+    "_tee_report_type_t",
+    "tee_mac_t",
+];
```
Comment (awygle) [inline tservice/sys/types/build.rs:21]: Should we be binding the other data types defined by `sgx_report2.h` here? It's only included by `sgx_util.h` and `internal/arch.h` from the SDK.

File: tservice/sys/wrapper.h [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#include <sgx_tseal.h>
```
Comment (awygle) [inline tservice/sys/wrapper.h:1]: Some functions exported by `tservice.a` are declared in `sgx_secure_align_api.h` (specifically `sgx_aligned_malloc`, `sgx_aligned_free`, and `sgx_get_aligned_ptr`. I believe this header should be included in the wrapper for that reason.

File: tservice/sys/types/build.rs (hunk 1 above)
Comment (awygle) [reply tservice/sys/types/build.rs:21]: There's also `align_req_t` defined in `sgx_secure_align_api.h`, which as I commented elsewhere declares certain functions from `tservice.a`.

File: tservice/sys/build.rs [hunk 3]
```diff
@@ -2,39 +2,58 @@
...
+    "sgx_unseal_data",
+    "sgx_mac_aadata",
+    "sgx_mac_aadata_ex",
+    "sgx_unmac_aadata",
+    "sgx_seal_data_iv",
+    "sgx_unseal_data_helper",
+    "sgx_calc_sealed_data_size",
+    "sgx_get_add_mac_txt_len",
+    "sgx_get_encrypt_txt_len",
+];
```
Comment (awygle) [inline tservice/sys/build.rs:36]: `sgx_seal_data_iv` and `sgx_unseal_data_helper` appear to be internal functions (declared in `sdk/tseal/tSeal_internal.h`)

Comment (awygle) [reply tservice/sys/build.rs:36]: `sgx_derive_target_from_report` appears to be an internal function (not declared in any header)

**Review by awygle — CHANGES_REQUESTED** [review]
The usual questions about what should be bound, and also a more direct suggestion about `sgx_secure_align_api.h`.

File: tservice/sys/wrapper.h (hunk 2 above)
Comment (nick-mobilecoin) [reply tservice/sys/wrapper.h:1]: added missing header

File: tservice/sys/build.rs (hunk 3 above)
Comment (nick-mobilecoin) [reply tservice/sys/build.rs:33]: removed private/internal functions from the bindings list

File: tservice/sys/types/build.rs (hunk 1 above)
Comment (nick-mobilecoin) [reply tservice/sys/types/build.rs:24]: Added the report 2 header so if/when we get to automating it "just works"

**Review by awygle — APPROVED** [review]

**Review by jcape — APPROVED** [review]

**Pull request merged**